### PR TITLE
Added file entry author

### LINKED ISSUE #78: File entries should show the the user who created them and date posted
## Description
NCO's are manually typing their name in the comments of file entries to indicate who posted the entry. We already store this information. File entries should list the person who created them and the date the were posted.

## Requirements/Expected Behavior
Update the file entry list view to show the date posted and user who created the entry.

### PR DESCRIPTION
**Pull request by MichaelJamesParsons**
Resolves #78 

Before this PR is pushed to prod, we need to run a migration on titan. The `modified_by` field in the `user_file_entries` table references legacy user IDs instead of titan's user IDs. This must be corrected to avoid errors.

    ```
    git rebase -i $(git merge-base HEAD master)
    git push origin NAME_OF_BRANCH --force-with-lease
    ```

### CODE REVIEW

**Comment by MichaelJamesParsons** [comment]
@ELD Ping

File: titan-server/src/accounts/file_entries.rs [hunk 1]
```diff
@@ -48,26 +33,53 @@ pub fn find_most_recent(
...
+) -> Result<models::UserFileEntryAssoc, diesel::result::Error> {
+    let user = accounts::users::find_by_id(file_entry.user_id, titan_db)?;
+    let user_profile = accounts::users::map_user_to_profile(
+        user, wcf_db, app_config)?;
+    let file_entry_type = accounts::file_entry_types::find_by_id(
+        file_entry.user_file_entry_type_id, titan_db)?;
+    let modified_by_user = accounts::users::find_by_id(file_entry.modified_by, titan_db)?;
+    let modified_by_profile = accounts::users::map_user_to_profile(
+        modified_by_user, wcf_db, app_config)?;
+
```
Comment (ELD) [inline titan-server/src/accounts/file_entries.rs]: nit: there appears to be some whitespace leftover here that we should trim.

File: titan-server/src/accounts/file_entries.rs [hunk 2]
```diff
@@ -48,26 +33,53 @@ pub fn find_most_recent(
...
+    app_config: &State<config::AppConfig>
+) -> Result<models::UserFileEntryAssoc, diesel::result::Error> {
+    let user = accounts::users::find_by_id(file_entry.user_id, titan_db)?;
+    let user_profile = accounts::users::map_user_to_profile(
+        user, wcf_db, app_config)?;
+    let file_entry_type = accounts::file_entry_types::find_by_id(
+        file_entry.user_file_entry_type_id, titan_db)?;
+    let modified_by_user = accounts::users::find_by_id(file_entry.modified_by, titan_db)?;
+    let modified_by_profile = accounts::users::map_user_to_profile(
+        modified_by_user, wcf_db, app_config)?;
```
Comment (ELD) [inline titan-server/src/accounts/file_entries.rs]: Any of the above failing results in no file entries being displayed. We should probably have a more graceful fallback to an unknown user or something of the like.

Comment (MichaelJamesParsons) [reply titan-server/src/accounts/file_entries.rs]: I understand your reasoning behind implementing a fallback, but I see that as a band aid over an underlying problem that will likely resurface again in another form. These associations must *always* be valid. Cases where they are not hint to corrupt records that should either be corrected or discarded.

**Comment by MichaelJamesParsons** [comment]
@ELD ping

**Comment by MichaelJamesParsons** [comment]
@ELD Ready for merge

**Pull request merged**